backend/app/models/token.py

```diff
@@ -118,13 +119,10 @@ def create_refresh_token(
         cls, user: User, device: str | None = None, oldRefreshToken: Self | None = None
     ) -> Tuple[str, Self]:
         assert device or oldRefreshToken
-        if oldRefreshToken and (
-            oldRefreshToken.type != "refresh"
-            or oldRefreshToken.has_created_refresh_token()
```

Comment:
Keep this but update the `has_created_refresh_token` like in the other method to only consider actually used tokens. Maybe rename it to `has_created_used_refresh_token`